test/performance/latency-controller/README.md

```diff
@@ -1,242 +1,156 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: nginx-test-deployment
-  namespace: default
-  labels:
-    app.kubernetes.io/name: nginx-singleton
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app: nginx
-  template:
-    metadata:
-      labels:
-        app: nginx
-    spec:
-      containers:
-      - name: nginx
-        image: nginx:alpine
-        ports:
-        - containerPort: 80
-        readinessProbe:
-          httpGet:
-            path: /
-            port: 80
-          initialDelaySeconds: 5
-          periodSeconds: 5
-EOF
-```
+5. **📝 Status Collection → WorkStatus Created (ITS)**
+   - The WEC cluster reports status back to the ITS cluster through a `WorkStatus` object.
 
-### Pods
+6. **✅ Status Propagation → WDS Object Status Updated**
+   - The collected status is propagated back and reflected in the original WDS object.
 
-Apply pods to generate observable latency metrics:
+## Features
 
-```bash
-# Pod 1
-kubectl --context "$wds_context" apply -f - <<EOF
-apiVersion: v1
-kind: Pod
-metadata:
-  name: nginx-pod
-  namespace: default
-  labels:
-    app.kubernetes.io/name: nginx-singleton
-spec:
-  containers:
-  - name: nginx
-    image: nginx:alpine
-    ports:
-    - containerPort: 80
-    readinessProbe:
-      httpGet:
-        path: /
-        port: 80
-      initialDelaySeconds: 5
-      periodSeconds: 5
-EOF
-```
+### Advanced Monitoring Capabilities
 
-```bash
-# Pod 2 (variation)
-kubectl --context "$wds_context" apply -f - <<EOF
-apiVersion: v1
-kind: Pod
-metadata:
-  name: nginx-pod-2
-  namespace: default
-  labels:
-    app.kubernetes.io/name: nginx-singleton
-spec:
-  containers:
-  - name: nginx
-    image: nginx:alpine
-    ports:
-    - containerPort: 80
-    readinessProbe:
-      httpGet:
-        path: /
-        port: 80
-      initialDelaySeconds: 5
-      periodSeconds: 5
-EOF
-```
+- **Multi-Cluster Support**: Monitors workloads across multiple WEC clusters simultaneously  
+- **Dynamic Resource Discovery**: Automatically discovers and monitors all configured Kubernetes resource types  
+ > **Note:** Currently, only **namespaced-scope resources** are supported. Cluster-scoped resources are not covered by this controller.
+- **Real-time Metrics**: Provides live latency measurements via Prometheus metrics  
+- **Workload Count Tracking**: Monitors the number of deployed workload objects per cluster  
+- **Status Field Detection**: Intelligently handles resources with and without status fields  
 
-## 🛠️ Build & Run
+## Installation
 
-Run
+### Step 1: Build and Push Controller Image
 
+Build and push the controller image to your container registry:
 ```bash
-go build -o bin/latency-controller ./test/performance/latency-controller
+Build the controller image
+make docker-copy
+make docker-build IMAGE=<your-registry>/latency-controller:tag
 
-./bin/latency-controller
+Push to registry
+make docker-push IMAGE=<your-registry>/latency-controller:tag
```

Comment:
Build and push your image to the location specified by `IMAGE` var:

First set your image:
```
IMAGE=<your-registry>/latency-controller:tag
```

Then: 

```
make docker-copy
make docker-build $IMAGE
make docker-push $IMAGE
```